zap/zap.go

```diff
@@ -0,0 +1,86 @@
+package zap
+
+import (
+	"errors"
+
+	"github.com/airbrake/gobrake/v5"
+	"go.uber.org/zap/zapcore"
+)
+
+type Core struct {
+	zapcore.LevelEnabler
+	Notifier   *gobrake.Notifier
+	coreFields map[string]interface{}
+	depth      int
+}
+
+func NewCore(enab zapcore.LevelEnabler, notifier *gobrake.Notifier) (*Core, error) {
+	if notifier == nil {
+		return nil, errors.New("airbrake notifier is not defined")
+	}
+	core := &Core{
+		LevelEnabler: enab,
+		Notifier:     notifier,
+		depth:        3,
+	}
+	return core, nil
+}
+
+// SetDepth method is for setting the depth of the notices
+func (core *Core) SetDepth(depth int) {
+	core.depth = depth
+}
+
+func (core *Core) With(fields []zapcore.Field) zapcore.Core {
+	clone := *core
```

Comment:
Why do we need to make a shallow copy and return it?

Reply:
I will remove it